packages/invoices/src/OutboundShipment/DetailView/DetailView.tsx

```diff
@@ -241,28 +244,36 @@ export const OutboundShipmentDetailViewComponent: FC = () => {
               color="secondary"
               onClick={() => navigate(-1)}
             />
-            <ButtonWithIcon
-              Icon={<DownloadIcon />}
-              labelKey="button.save"
-              variant="contained"
-              color="secondary"
-              onClick={() => {
-                success('Saved invoice! 🥳 ')();
-                save(draft);
-              }}
-            />
-            <ButtonWithIcon
-              Icon={<ArrowRightIcon />}
-              labelKey="button.save-and"
-              labelProps={{ status: draft.status }}
-              variant="contained"
-              color="secondary"
-              onClick={() => {
-                success('Saved invoice! 🥳 ')();
-                draft.update?.('status', 'finalised');
-                save(draft);
-              }}
-            />
+            {isInvoiceSaveable(draft) && (
+              <>
+                <ButtonWithIcon
+                  Icon={<DownloadIcon />}
+                  labelKey="button.save"
+                  variant="contained"
+                  color="secondary"
+                  onClick={() => {
+                    success('Saved invoice! 🥳 ')();
+                    save(draft);
+                  }}
+                />
+                <ButtonWithIcon
+                  Icon={<ArrowRightIcon />}
+                  labelKey="button.save-and-confirm-status"
+                  labelProps={{
+                    status: t(
+                      getNextOutboundStatusButtonTranslation(draft.status)
+                    ),
+                  }}
+                  variant="contained"
+                  color="secondary"
+                  onClick={() => {
+                    success('Saved invoice! 🥳 ')();
+                    draft.update?.('status', 'finalised');
+                    save(draft);
```

Comment:
So I'm not really sure if we should be showing these, or not or what to show so just did this for now